docs/enhanced-provider-k8s.md

```diff
@@ -0,0 +1,303 @@
+
+### Generic References
+
+To address such a problem, we can leverage the idea of **Generic References** which is currently a [design proposal](https://github.com/crossplane/crossplane/pull/2385) under discussion in Crossplane community.
+
+As it states in the design document, it is aimed to address the scenario where managed resource requires information that is not available when user creates the resource. For example, to create a Subnet, you need to supply a VPC ID. If you create the VPC at the same time, then the ID will be available only after the creation.
+
+According to the proposal, all managed resources will have a top-level spec field that lists the generic references. They will be resolved in order and failure of any one will cause reconciliation to fail.
+
+Below is a sample snippet for a managed resource which has a reference to another managed resource:
+
+```yaml
+spec:
+  patches:
+  - fromObject:
+      apiVersion: ec2.aws.crossplane.io/v1alpha1
+      kind: VPC
+      name: main-vpc # optional, matchLabels can be used instead.
+      fieldPath: spec.forProvider.cidrBlock
+    toFieldPath: spec.cidrBlock
+```
+
+And, this one is a reference to a namespaced Kubernetes resource.
+
+```yaml
+spec:
+  patches:
+  - fromObject:
+      apiVersion: v1
+      kind: ConfigMap
+      name: common-settings
+      namespace: crossplane-system
+      fieldPath: data.region
+    toFieldPath: spec.forProvider.region
+```
+
+Although this design has not been closed yet, a similar idea has been implemented in the enhanced provider. When define an `Object` resource, it allows you to specify one or more references in `spec.references` using exactly the same syntax as it is defined in the above design.
+
+In our case, when define `Object` for `ClusterServiceVersion` resource, you need to define a reference to `Subscription` so that the `ClusterServiceVersion` name can be retrieved from `Subscription` at the field path `status.atProvider.manifest.status.currentCSV` when it is created, then apply the value to `Object` as the `ClusterServiceVersion` name at `spec.forProvider.manifest.metadata.name`:
+
+```yaml
+apiVersion: kubernetes.crossplane.io/v1alpha1
+kind: Object
+metadata:
+  name: csv-kong
+  annotations:
+    kubernetes.crossplane.io/managementType: "ObservableAndDeletable"
+spec:
+  references:
+  - fromObject:
+      apiVersion: kubernetes.crossplane.io/v1alpha1
+      kind: Object
```

Comment:
Totally understood. We should limit the scope based on provider-kubernetes original design.

Or, we could limit to Object type by default and give people choice to extend it intentionally on a case by case basis. To reference arbitrary resources aligns with what is proposed in [Generic References design](https://github.com/crossplane/crossplane/pull/2385).

In that case, people MUST configure the necessary RBAC by themselves explicitly and clearly understand what that means. I think provider-kubernetes support both in-cluster and outside cluster case. It may give the provider wider range of potential for its usage scenarios.

Reply:
Ok, after a second though, this sounds good. 
I believe, people would still need to explicitly add more permissions in Generic references design and we are in a similar situation here 👍